evm/src/cpu/kernel/asm/mpt/hash/hash_trie_specific.asm

```diff
@@ -1,272 +1,312 @@
     SWAP5 POP
-    // stack: rlp_pos, num_topics, i, num_logs, data_len_ptr, old_rlp_pos, payload_len_ptr, retdest
+    // stack: rlp_pos, num_topics, i, num_logs, data_len_ptr, old_rlp_pos, payload_len_ptr, cur_len', retdest
     SWAP5 POP
-    // stack: num_topics, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: num_topics, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, cur_len', retdest
     POP
-    // stack: i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, cur_len', retdest
     // Encode data prefix.
     DUP3 %mload_trie_data
-    // stack: data_len, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: data_len, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, cur_len', retdest
+
+    // Add `data_len` to the length of the trie data.
+    DUP1 SWAP7 ADD SWAP6
+
+    // stack: data_len, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     DUP4 %increment DUP2 ADD
-    // stack: next_log_ptr, data_len, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: next_log_ptr, data_len, i, num_logs, data_len_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     SWAP4 %increment
-    // stack: data_ptr, data_len, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: data_ptr, data_len, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     PUSH @SEGMENT_TRIE_DATA PUSH 0
-    // stack: SRC, data_len, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: SRC, data_len, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     DUP8
-    // stack: rlp_pos, SRC, data_len, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: rlp_pos, SRC, data_len, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     %encode_rlp_string
-    // stack: new_rlp_pos, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: new_rlp_pos, i, num_logs, next_log_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     SWAP4 POP
-    // stack: i, num_logs, next_log_ptr, new_rlp_pos, payload_len_ptr, retdest
+    // stack: i, num_logs, next_log_ptr, new_rlp_pos, payload_len_ptr, cur_len'', retdest
     %increment
     %jump(encode_receipt_logs_loop)
 
 encode_receipt_end:
-    // stack: num_logs, num_logs, current_log_ptr, rlp_pos, payload_len_ptr, retdest
+    // stack: num_logs, num_logs, current_log_ptr, rlp_pos, payload_len_ptr, cur_len'', retdest
     %pop3
-    // stack: rlp_pos, payload_len_ptr, retdest
+    // stack: rlp_pos, payload_len_ptr, cur_len'', retdest
     SWAP1 POP
-    // stack: rlp_pos, retdest
-    SWAP1
+    // stack: rlp_pos, cur_len'', retdest
+    %stack(rlp_pos, new_len, retdest) -> (retdest, rlp_pos, new_len)
     JUMP
 
 encode_nonzero_receipt_type:
-    // stack: txn_type, rlp_pos, value_ptr, retdest
```

Comment:
minor, but I'd keep a stack description when reaching this label for extra clarity (even though the next sequence of operations is fairly simple to follow)